### PR TITLE
fix void-packages issue #732

### PR DESCRIPTION
**Pull request by pulux**
parse the passfile and options in /etc/crypttab, need mor test functions.

### CODE REVIEW

File: crypt.awk [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+            system("mkswap /dev/mapper/" $1 );
+        }
+        else if ( maketmp == "y" )
+        {
+            system("cryptsetup " cmd " -d " $3 " create " src " " $1"_unformatted");
+            system("mkefs -t ext4 -q /dev/mapper/" $1"_unformatted" );
+            mdir="/run/cryptsetup/" $1;
+            system("mkdir -p " mdir);
+            system("mount /dev/mapper/" $1 "_unformatted  " mdir);
+            system("chmod 1777 " mdir);
```
Comment (xtraeme) [inline crypt.awk]: Seriously? this is probably not a good idea.

**Comment by pulux** [comment]
Why do you think so? Is it better don't support key-files, key-scripts and other options and manually input the passwords for all drives in a n disk-array?

What should be the right way to parse the crypttab file?

**Comment by xtraeme** [comment]
My comment was only related to `chmod 1777`, not about the whole awk script.

**Comment by pulux** [comment]
The script functions are based on the fedora crypto.functions and the debian cryptdisk.functions and need more work in detail.

What functions are necessary in a first stage? I need keyfile and keyscript support.

**Comment by pulux** [comment]
i have make cleanup the script and opened an new pull-request #12. can you pleas check and hopfully merge. Thanks.

**Pull request merged**